src/replication/core/controller.go

```diff
@@ -118,15 +118,34 @@ func (ctl *Controller) UpdatePolicy(updatedPolicy models.ReplicationPolicy) erro
 		return fmt.Errorf("policy %d not found", id)
 	}
 
-	if err = ctl.triggerManager.UnsetTrigger(id, *originPolicy.Trigger); err != nil {
-		return err
+	reset := false
+	if updatedPolicy.Trigger.Kind != originPolicy.Trigger.Kind {
+		reset = true
+	} else {
+		switch updatedPolicy.Trigger.Kind {
+		case replication.TriggerKindSchedule:
+			if updatedPolicy.Trigger.Param != originPolicy.Trigger.Param {
+				reset = true
+			}
+		case replication.TriggerKindImmediate:
+			// Always reset immediate trigger as it is relevent with namespaces
+			reset = true
+		default:
+			// manual trigger, no need to reset
+		}
 	}
 
-	if err = ctl.policyManager.UpdatePolicy(updatedPolicy); err != nil {
-		return err
+	if reset {
+		if err = ctl.triggerManager.UnsetTrigger(id, *originPolicy.Trigger); err != nil {
+			return err
+		}
+		if err = ctl.policyManager.UpdatePolicy(updatedPolicy); err != nil {
```

Comment:
Only need to call one update policy function and after that calling unset and setup to update the trigger